File: java/src/hackerrank/Camelcase.java
package hackerrank;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;


// https://www.hackerrank.com/challenges/camelcase/problem

public class Camelcase {

	public static int camelcase(String s) {
		if (s == null || s.isEmpty()) return 0;
		long count = s.chars().filter(Character::isUpperCase).count();
		return (int) count + 1;
    }

	
	@Test
	@DisplayName("Test sample 0")
	public void testInput0() {
		String s = "saveChangesInTheEditor";
		int expected = 5;
		int result = camelcase(s);
		assertEquals(expected, result);
	}
	
}

File: kotlin/src/hackerrank/camelcase.kt
package hackerrank

import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

// https://www.hackerrank.com/challenges/camelcase/problem

fun camelcase(s: String): Int {
    if (s.isNullOrBlank()) return 0
    var count = s.toCharArray().filter { it.isUpperCase() }.count()
    return count + 1
}


internal class CamelcaseTest {

    @Test
    @DisplayName("Test sample input 0")
    fun test() {
        val s = "saveChangesInTheEditor"
        val expected: Int = 5
        val result = camelcase(s)
        assertEquals(expected, result)
    }

}
